pipeline/vars/common.groovy

```diff
@@ -0,0 +1,127 @@
+#!/usr/bin/env groovy
+/*
+    Common groovy methods that can be reused by the pipeline jobs.
+*/
+
+import groovy.json.JsonSlurper
+
+String composeId = ""
+String instanceName = ""
+
+def prepareNode() {
+    /*
+        Installs the required packages needed by the Jenkins node to
+        run and execute the cephci test suites.
+    */
+    sh '''
+        #!/bin/bash
+
+        echo "Initialize Node"
+        sudo yum install -y wget git-core python3
+
+        # Workaround: Disable IPv6 to have quicker downloads
+        sudo sysctl -w net.ipv6.conf.eth0.disable_ipv6=1
+
+        # Mount reesi for storing logs
+        if [ ! -d "/ceph" ]; then
+            echo "Mounting ressi004"
+            sudo mkdir -p /ceph
+            sudo mount -t nfs -o sec=sys,nfsvers=4.1 reesi004.ceph.redhat.com:/ /ceph
+        fi
+
+        # Copy the auth files from internal server
+        wget http://magna002.ceph.redhat.com/cephci-jenkins/.osp-cred-ci-2.yaml -O ${HOME}/osp-cred-ci-2.yaml
+        wget http://magna002.ceph.redhat.com/cephci-jenkins/.cephci.yaml -O ${HOME}/.cephci.yaml
+
+        # Install cephci pre-requisistes
+        rm -rf .venv
+        python3 -m venv .venv
+        source .venv/bin/activate
+        python -m pip install --upgrade pip
+        python -m pip install -r requirements.txt
+        deactivate
+    '''
+}
+
+def runTestSuite() {
+    /*
+        Execute the test suite by processing CI_MESSAGE for container
+        information along with the environment variables. The required
+        variables are
+            - sutVMConf
+            - sutConf
+            - testSuite
+            - addnArgs
+    */
+    println "Begin Test Suite execution"
+
+    // generate random instance name
+    def sec = sh(script: "echo \$(date +%s)", returnStdout: true).trim()
+    instanceName = "psi-${sec}"
+
+    def rhceph_ver = "${env.rhceph_ver}" ?: '5.0-rhel-8'
+
+    // Build the CLI options
+    def cmd = "${env.WORKSPACE}/.venv/bin/python run.py"
+    cmd += " --osp-cred ${env.HOME}/osp-cred-ci-2.yaml"
+    cmd += " --instances-name ${instanceName}"
+
+    // Append test suite specific
+    cmd += " --inventory ${env.sutVMConf}"
+    cmd += " --global-conf ${env.sutConf}"
+    cmd += " --suite ${env.testSuite}"
+
+    // If CI_MESSAGE is populated, the preference is use the information.
+    def ciMessage = "${params.CI_MESSAGE}" ?: ""
+    if (ciMessage?.trim()) {
```

Comment:
what is the probability of not getting CI message details, can we raise build error in case not getting CI_Message.

Reply:
The method being common and to support re-usability... it's possible that the `ciMessage` is not being set.

Reply:
makes sense. However in this case I think we may need to add docker image run-time options to go with right ceph-compose and ceph images, if CI_Message is not set.

Reply:
Good question, that is why I have inserted `addnArgs` environment variable. :)